rust/operator-binary/src/listener_controller.rs

```diff
@@ -217,23 +220,27 @@ pub async fn reconcile(listener: Arc<Listener>, ctx: Arc<Ctx>) -> Result<control
     let ports: BTreeMap<String, i32>;
     match listener_class.spec.service_type {
         ServiceType::NodePort => {
-            let endpoints = ctx
+            let pvs = ctx
                 .client
-                .get_opt::<Endpoints>(&svc_name, ns)
+                .list_with_label_selector::<PersistentVolume>(
+                    &(),
+                    &LabelSelector {
+                        match_labels: Some(listener_persistent_volume_label(&listener).unwrap()),
+                        ..Default::default()
+                    },
+                )
                 .await
-                .with_context(|_| GetObjectSnafu {
-                    obj: ObjectRef::<Endpoints>::new(&svc_name).within(ns).erase(),
-                })?
-                // Endpoints object may not yet be created by its respective controller
-                .unwrap_or_default();
-            let node_names = endpoints
-                .subsets
+                .unwrap();
```

Comment:
This will only work with volumes that have been mounted at least once since upgrading.. maybe we want to merge the two mechanisms to help make migration more seamless?

Reply:
a4a8cfab236f28b59ca0622ad50301509c75b28c